Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED

I've been attempting to install CPPRestSDK by running .\vcpkg install cpprestsdk cpprestsdk:x64-windows, but I always get an error while building cpprestsdk:x86-windows. The error and logs are in the block below
My actual username in the log has been replaced with [My User] for privacy

```
 The following packages will be built and installed:

  * boost-algorithm[core]:x64-windows
  * boost-asio[core]:x64-windows
  * boost-atomic[core]:x64-windows
  * boost-bind[core]:x64-windows
  * boost-concept-check[core]:x64-windows
  * boost-container[core]:x64-windows
  * boost-context[core]:x64-windows
  * boost-conversion[core]:x64-windows
  * boost-coroutine[core]:x64-windows
  * boost-date-time[core]:x64-windows
  * boost-exception[core]:x64-windows
  * boost-function[core]:x64-windows
  * boost-function-types[core]:x64-windows
  * boost-functional[core]:x64-windows
  * boost-fusion[core]:x64-windows
  * boost-io[core]:x64-windows
  * boost-iterator[core]:x64-windows
  * boost-lambda[core]:x64-windows
  * boost-lexical-cast[core]:x64-windows
  * boost-math[core]:x64-windows
  * boost-numeric-conversion[core]:x64-windows
  * boost-optional[core]:x64-windows
  * boost-pool[core]:x64-windows
  * boost-range[core]:x64-windows
  * boost-regex[core]:x64-windows
  * boost-smart-ptr[core]:x64-windows
  * boost-thread[core]:x64-windows
  * boost-tokenizer[core]:x64-windows
  * boost-tuple[core]:x64-windows
  * boost-type-index[core]:x64-windows
  * boost-unordered[core]:x64-windows
  * brotli[core]:x64-windows
    cpprestsdk[brotli,core,default-features,websockets]:x86-windows
    cpprestsdk[brotli,core,default-features,websockets]:x64-windows
  * openssl[core]:x64-windows
  * openssl-windows[core]:x64-windows
  * websocketpp[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/38: cpprestsdk:x86-windows
Building package cpprestsdk[brotli,core,default-features,websockets]:x86-windows...
-- Using cached C:/Users/[My User]/CPPLibs/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
-- Using source at C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:202 (message):
    Command failed: C:/Program Files/CMake/bin/cmake.exe;--build;.;--config;Debug;--target;install;--;-v
    Working Directory: C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/x86-windows-dbg
    See logs for more information:
      C:\Users\[My User]\CPPLibs\vcpkg\buildtrees\cpprestsdk\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/cpprestsdk/portfile.cmake:43 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: cpprestsdk:x86-windows
  Vcpkg version: 2018.11.23-nohash
```

[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/3170389/install-x86-windows-dbg-out.log)
This is the log mentioned in the output that should contain extra information.

Yeah, I googled the command and found that,  but it still didn't fix the install, still getting this

```
The following packages will be built and installed:
  * boost-asio[core]:x64-windows
  * brotli[core]:x64-windows
    cpprestsdk[brotli,core,default-features,websockets]:x86-windows
    cpprestsdk[brotli,core,default-features,websockets]:x64-windows
  * openssl[core]:x64-windows
  * openssl-windows[core]:x64-windows
  * websocketpp[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/7: cpprestsdk:x86-windows
Building package cpprestsdk[brotli,core,default-features,websockets]:x86-windows...
-- Using cached C:/Users/[My User]/CPPLibs/vcpkg/downloads/Microsoft-cpprestsdk-v2.10.13.tar.gz
-- Using source at C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/src/v2.10.13-fe1b687772
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:56 (message):
    Command failed: ninja;-v
    Working Directory: C:/Users/[My User]/CPPLibs/vcpkg/buildtrees/cpprestsdk/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Users\[My User]\CPPLibs\vcpkg\buildtrees\cpprestsdk\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:273 (vcpkg_execute_required_process)
  ports/cpprestsdk/portfile.cmake:29 (vcpkg_configure_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package cpprestsdk:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: cpprestsdk:x86-windows
  Vcpkg version: 2018.11.23-nohash

Additionally, attach any relevant sections from the log files above.
```

--- a/scripts/buildsystems/vcpkg.cmake
+++ b/scripts/buildsystems/vcpkg.cmake
@@ -130,6 +130,7 @@ else() #Release build: Put Release paths before Debug paths. Debug Paths are req
 endif()
 
 file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _programfiles)
+file(TO_CMAKE_PATH "$ENV{PROGRAMFILES(x86)}" _programfiles_x86)
 set(CMAKE_SYSTEM_IGNORE_PATH
     "${_programfiles}/OpenSSL"
     "${_programfiles}/OpenSSL-Win32"
@@ -138,6 +139,13 @@ set(CMAKE_SYSTEM_IGNORE_PATH
     "${_programfiles}/OpenSSL-Win64/lib/VC"
     "${_programfiles}/OpenSSL-Win32/lib/VC/static"
     "${_programfiles}/OpenSSL-Win64/lib/VC/static"
+    "${_programfiles_x86}/OpenSSL"
+    "${_programfiles_x86}/OpenSSL-Win32"
+    "${_programfiles_x86}/OpenSSL-Win64"
+    "${_programfiles_x86}/OpenSSL-Win32/lib/VC"
+    "${_programfiles_x86}/OpenSSL-Win64/lib/VC"
+    "${_programfiles_x86}/OpenSSL-Win32/lib/VC/static"
+    "${_programfiles_x86}/OpenSSL-Win64/lib/VC/static"
     "C:/OpenSSL/"
     "C:/OpenSSL-Win32/"
     "C:/OpenSSL-Win64/"

